lib/dpn/client/agent/connection.rb

```diff
@@ -127,50 +139,32 @@ def request(method, url, query, body, &block)
               body: body.to_json,
               follow_redirect: true
           }
-
-          logger.info("Sending #{method.upcase}: #{File.join(base_url, fix_url(url))} #{options} ")
-          raw_response = connection.request method, fix_url(url), options
+          logger.info("Sending #{method.upcase}: #{File.join(base_url, url)} #{options} ")
+          raw_response = connection.request method, url, options
           logger.debug("Received #{raw_response.inspect}")
           response = DPN::Client::Response.new(raw_response)
           logger.info("Received #{response.status}")
           if block_given?
             yield response
           end
-
-          return response
+          response
         end
 
-
         def parse_url(raw_url)
           url, query = raw_url.split("?", 2)
-          url = File.join url, ""
-          if query
-            query = URI::decode_www_form(query).to_h
-          else
-            query = {}
-          end
-          return url, query
+          query = if query
+                    URI::decode_www_form(query).to_h
+                  else
+                    {}
+                  end
+          return fix_url(url), query
         end
 
-
-        def connection
-          @connection ||= ::HTTPClient.new({
-              agent_name: user_agent,
-              base_url: base_url,
-              default_header: {
-                  "Content-Type" => "application/json",
-                  "Authorization" => "Token #{auth_token}"
-              },
-              force_basic_auth: true
-          })
-        end
-
-
         def fix_url(url)
-          File.join url, "/"
+          url = File.join(api_ver, url) unless url.include?(api_ver)
+          File.join "/", url, "/"
```

Comment:
Great question and raises concerns about slash-duplication.  Not clear about a precise answer to this and whether it interacts with how the httpclient constructs and maintains the URIs.  Ideally, all of the URI handling might be better using a standard `URI` library, but that's not exactly the goal of this PR (might be spin-off issue).  Anyhow, testing this in dpn-sync could be sufficient to know whether or not to use the slash-prefix here.